frb_codegen/src/generator/dart/ty_struct.rs

```diff
@@ -57,14 +59,17 @@ impl TypeDartGeneratorTrait for TypeStructRefGenerator<'_> {
             inner.insert(0, "bridge: bridge,".to_string());
         }
         let inner = inner.join("\n");
-
+        let cast = "final arr = raw as List<dynamic>;".to_string();
+        let safe_check = if self.is_exception {
+            "".to_string()
+        } else {
+            format!("if (arr.length != {}) throw Exception('unexpected arr length: expect {} but see ${{arr.length}}');", s.fields.len(), s.fields.len())
+        };
```

Comment:
Well I am not very sure - why we can skip such error checks when it is an exception?